docs/storage-config.md

```diff
@@ -0,0 +1,267 @@
+* Copy-on-read: an overlay is provided with copy-on-read semantics against the
+  master storage. The local overlay will contain both changes made by the guest,
+  and copies of any blocks read.
+* Clone: A full copy of the master storage is made to a local file. Once this is
+  created, no further use is made of the master storage.
+
+The first copy mode provides long term persistent storage, while the other modes
+provide ephemeral storage that is discarded when the virtual machine is deleted
+from the node.
+
+The host needs to have somewhere to store the local overlays / files. This is
+achieved by configuring one or more volumes for the libvirtd POD when it is
+created, and mounting them into a defined location. Any valid PVC is usable for
+this purpose. If the PVC backing store is only accessible for a single host,
+then when performing migration, QEMU block storage migration feature will need
+to be enabled. If the PVC references a networked filesystem that can be mounted
+on multiple hosts, then VMs can migrate with no need to copy emphemeral disks.
+If multiple volume mounts are made available to the libvirtd POD, annotations
+would need to be used to tag them for different usage types. This might be useful
+if a cluster needs to provide ephemeral storage with differing performance grades
+or costs (eg one volume provides cheap & slow rotating rust, while another volume
+provides expensive & fast solid state). The VM spec disk configuration would be
+able to optionally include a reference to the usage type it desires, to allow
+virt-handler to map the disk to the particular volume mount.
+
+Thus the libvirtd container may contain
+
+```yaml
+    annotations:
+      kubevirt.io/storage.usage-type.fast: /storage/ssd
+      kubevirt.io/storage.usage-type.slow: /storage/rust
+    volumeMounts:
+      - mountPath: /storage/ssd
+        name: storage-ssd
+      - mountPath: /storage/rust
+        name: storage-rust
+```
+
+NB The DaemonSet resource does not provide a way to dynamically assign
+different PVCs to each libvirtd POD it creates. Thus in the short term, to make
+use of this feature would require the cloud admin to manually create libvirtd
+POD specs for each compute node, instead of using a DeamonSet resource. It is
+anticipated that either DaemonSet would be enhanced eventually, or replaced by
+a more advanced resource that can dynamically assigns PVCs to each POD.
+
+When configuring the VM spec disk
+
+```yaml
+    disks:
+      - source:
+          pvc:
+	    claimname: some-pvc-name
+	access-mode: read-only
+	copy:
+	  mode: copy-on-write
+          usage-type: fast
+```
+
+The 'usage-type' attribute refers to the 'kubevirt.io/sotrage.usage-type'
+annotations provided by the libvirtd POD. Either all libvirtd PODs would need
+to support the same annotations, or the scheduler will need to be able to only
+place VMs on the nodes where the libvirtd POD has the neccessary annotations.
```

Comment:
Such annotations can be placed on the node directly, no need to place them on a specific pod. The scheduler can take them into account. VMs have a `nodeSelector` section, which is mapped to the `nodeSelector` section of a Pod. That is taken into account.